### DISCUSSION TITLE
Thoughtbot library migration

### DISCUSSION THREAD

**Discussion by doomspork** [beam-community/beam-community.org#22] [General]
I want to shift these conversations from email into the open where we can engage a broader slice of the community in these discussions which have impact beyond just the maintainers (previous and current).

Both @yordis and @btkostner have expressed interest in helping with the migration but I welcome other's engagement. @brimatteng from Thoughtbot has helped with the transfer but there remain a few things for us to complete on our end:

1. Transfer hex ownership of the packages
2. Update the CI workflows
3. Put together BEAM Community contribution guidelines (look to Thoughtbot for examples)
4. Publish a blog post announcing the completed transfer, directing folks to the new projects, and the blog post
5. Work through open issues/PRs

Longer term, and perhaps a separate issue, we can discuss the long term plans for these libraries and others under the umbrella.

cc @mattpolzin

**Comment by doomspork** [comment]
I welcome feedback from all members of @beam-community/team, the more the merrier!

**Reply by bglusman** [reply]
I assume this is in reference to https://github.com/beam-community/ex_machina and not any other thoughtbot elixir libraries here? https://github.com/orgs/thoughtbot/repositories?q=&type=all&language=elixir&sort=

Not sure if I can help much, but thought context might help for anyone not caught up on scope of conversation, including myself

**Reply by doomspork** [reply]
Thanks @bglusman and apologized for not including more context. This actually relates to all of the libraries, not just `ex_machina`.

**Reply by bglusman** [reply]
Ahh, now I see, I hadn't seen this announcement https://thoughtbot.com/blog/changing-maintainers-for-elixir-projects thanks Sean/good to know!

**Comment by yordis** [comment]
> Transfer Hex Ownership of Packages

I'm keen on drafting a guide for this process. I can assist effectively given access to the repository in question, and optimally, the Beam Community Hex Organization. Though I am willing to delegate tasks, my concern lies in identifying which `HEX_TOKEN` is specified in the Actions Secrets, since I focus on CI/CD tasks right away to stream line collaboration.

> Update CI Workflows

In an effort to foster accountability within the Elixir ecosystem, I've engaged in discussions on [this thread](https://github.com/erlef/libs-and-frameworks/discussions/3). My experiences with Tesla, Ueberauth, BEAM Community, and my own packages have provided insight into the dos and don'ts. I will share more insights as I refine processes.

> Establish BEAM Community Contribution Guidelines (refer to Thoughtbot for inspiration)

Having spearheaded a similar initiative in Ueberauth ([see here](https://github.com/ueberauth/.github)), though not flawless, it's a step forward. I aim to extend similar guidelines here, and am exploring the creation of ecosystem-wide guidelines. Your collaboration in refining and disseminating this content would be invaluable.

> Address Open Issues/PRs

Progressively tackling these! Just balancing it with maintenance of other packages.

**Comment by brian-penguin** [comment]
Oh my! I'm sorry I've been on a client locked down laptop for months and didn't realize the hex transfer had failed to run. I've fixed it. Is someone available to check that Ueberauth on hex.pm now owns bamboo and ex_machina?

**Reply by yordis** [reply]
@brimatteng I got a notification that I have ownership now, so, I think so.
